src/Credit/Credit.cpp

```diff
@@ -28,18 +28,34 @@ struct CreditWidget : ModuleWidget {
     ModuleWriter mWriter;
 
     static void saveModules(ModuleWriter& mWriter) {
-        auto pathC = file_utils::getChosenFilePath();
+#ifdef USING_CARDINAL_NOT_RACK
+        std::string dir, ignore;
+        file_utils::getDefaultFilePath(dir, ignore);
+        async_dialog_filebrowser(true, dir.c_str(), "Save credit file", [&mWriter](char* pathC) {
+            if (pathC == nullptr) {
+                return; // fail silently
+            }
+            saveModulesPath(mWriter, pathC);
+            std::free(pathC);
```

Comment:
Since this logic is shared by both branches of the `#ifdef`, maybe we could refactor it into a function or lambda, to avoid the repetition.

Reply:
that would be nice. the repetition is there because your current usage assumes the current osdialog blocking behaviour.
I tried to make the patch as small as possible, but if you are ok with a little reimplementation that would be cool too